### PR TITLE
Feature/delivery extensions run setup and execution

### PR DESCRIPTION
**Pull request by lmatoti**
Summary of this change:
Delivery Extensions
Removing legacy data from run setups/executions

### CODE REVIEW

File: lib/messcada/services/can_tip_bin.rb [hunk 1]
```diff
@@ -123,61 +125,54 @@ def check_valid_bin_for_kromco_rmt_system # rubocop:disable Metrics/AbcSize
...
     end
 
-    def legacy_check_track_indicator_code
-      # log_code_check('Track indicator', run_criteria['track_indicator_code'], rmt_bin[:legacy_data]['track_slms_indicator_1_code'])
-      return if run_criteria['track_indicator_code'] == rmt_bin[:legacy_data]['track_slms_indicator_1_code']
+    def bintip_criteria_check_rmt_code
+      # log_code_check('Rmt Code', run_attrs[:rmt_code], rmt_bin[:rmt_code])
+      return if rmt_bin[:rmt_code_id] == run_attrs[:rmt_code_id]
+
+      legacy_errors << format(LEGACY_ERROR_MSG, 'Rmt Code', run_attrs[:rmt_code] || 'blank', rmt_bin[:rmt_code] || 'blank')
```
Comment (jcsjcs) [inline lib/messcada/services/can_tip_bin.rb:161]: Rmt should be RMT

File: lib/messcada/services/can_tip_bin.rb [hunk 2]
```diff
@@ -123,61 +125,54 @@ def check_valid_bin_for_kromco_rmt_system # rubocop:disable Metrics/AbcSize
...
 
-    def legacy_check_ripe_point_code
-      # log_code_check('Ripe point', run_criteria['ripe_point_code'], rmt_bin[:legacy_data]['ripe_point_code'])
-      return if run_criteria['ripe_point_code'] == rmt_bin[:legacy_data]['ripe_point_code']
+    def bintip_criteria_check_actual_ripeness_treatment
+      # log_code_check('Ripe point', run_attrs[:actual_ripeness_treatment_code], rmt_bin[:actual_ripeness_treatment_code])
+      return if rmt_bin[:actual_ripeness_treatment_id] == run_attrs[:actual_ripeness_treatment_id]
 
-      legacy_errors << format(LEGACY_ERROR_MSG, 'Ripe point', run_criteria['ripe_point_code'] || 'blank', rmt_bin[:legacy_data]['ripe_point_code'] || 'blank')
+      legacy_errors << format(LEGACY_ERROR_MSG, 'Ripe point', run_attrs[:actual_ripeness_treatment_code] || 'blank', rmt_bin[:actual_ripeness_treatment_code] || 'blank')
```
Comment (jcsjcs) [inline lib/messcada/services/can_tip_bin.rb:175]: Are cold treatment and ripeness both known as `Ripe point`?

**Review by jcsjcs — CHANGES_REQUESTED** [review]

**Comment by lmatoti** [comment]
Code review changes implemented

File: lib/messcada/repositories/messcada_repo.rb [hunk 3]
```diff
@@ -642,5 +642,44 @@ def find_rmt_size_id_for(fruit_size_reference_id)
...
+    def for_select_rmt_codes_by_run_cultivar(production_run_id, cultivar_id)
+      query = <<~SQL
+        SELECT "rmt_code", "rmt_codes"."id"
+        FROM "production_runs"
+        INNER JOIN "rmt_variants" ON ("rmt_variants"."cultivar_id" = "production_runs"."cultivar_id")
+        INNER JOIN "rmt_codes" ON ("rmt_codes"."rmt_variant_id" = "rmt_variants"."id")
+        WHERE ("production_runs"."id" = ? and "production_runs"."cultivar_id" = ?)
+      SQL
+      DB[query, production_run_id, cultivar_id].select_map(%i[rmt_code id])
+    end
```
Comment (jcsjcs) [inline lib/messcada/repositories/messcada_repo.rb:670]: This has the same pattern as `for_select_run_colour_percentages` - same change applies.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

**Comment by lmatoti** [comment]
Code review changes implemented

**Pull request merged**